internal/exec/oci_utils.go

```diff
@@ -102,6 +125,463 @@ func pullImage(ref name.Reference) (*remote.Descriptor, error) {
+	}
+	if subscriptionID == "" {
+		return nil, fmt.Errorf("AZURE_SUBSCRIPTION_ID environment variable is required")
+	}
+
+	log.Debug("Authenticating with Azure Container Registry", "registry", registry, "acrName", acrName)
+
+	// For now, return an error indicating this needs Azure SDK implementation
+	// In a full implementation, you would:
+	// 1. Create Azure credential using azidentity.NewClientSecretCredential
+	// 2. Create ACR client using armcontainerregistry.NewClient
+	// 3. Call GetCredentials to get username/password
+	return nil, fmt.Errorf("Azure ACR authentication requires Azure SDK implementation - ACR name: %s", acrName)
+}
+
+// hasAzureCLI checks if Azure CLI is available
+func hasAzureCLI() bool {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "az", "version")
+	return cmd.Run() == nil
+}
+
+// getACRAuthViaCLI attempts to get ACR credentials using Azure CLI
+func getACRAuthViaCLI(registry string) (authn.Authenticator, error) {
+	// Extract ACR name from registry URL
+	acrName := ""
+	if strings.HasSuffix(registry, ".azurecr.io") {
+		acrName = strings.TrimSuffix(registry, ".azurecr.io")
+	} else {
+		return nil, fmt.Errorf("invalid Azure Container Registry format: %s (expected <name>.azurecr.io)", registry)
+	}
+
+	if acrName == "" {
+		return nil, fmt.Errorf("could not extract ACR name from registry: %s", registry)
+	}
+
+	// Use Azure CLI to get ACR credentials
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "az", "acr", "credential", "show", "--name", acrName)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ACR credentials via Azure CLI: %w", err)
+	}
+
+	// Parse the JSON output
+	var result struct {
+		Passwords []struct {
+			Name  string `json:"name"`
+			Value string `json:"value"`
+		} `json:"passwords"`
+		Username string `json:"username"`
+	}
+
+	if err := json.Unmarshal(output, &result); err != nil {
+		return nil, fmt.Errorf("failed to parse Azure CLI output: %w", err)
+	}
+
+	if result.Username == "" {
+		return nil, fmt.Errorf("no username returned from Azure CLI")
+	}
+
+	if len(result.Passwords) == 0 {
+		return nil, fmt.Errorf("no passwords returned from Azure CLI")
+	}
+
+	// Use the first password (usually there are two - one for each credential)
+	password := result.Passwords[0].Value
+
+	log.Debug("Successfully obtained ACR credentials via Azure CLI", "registry", registry, "acrName", acrName)
+
+	return &authn.Basic{
+		Username: result.Username,
+		Password: password,
+	}, nil
+}
+
+// getGCRAuth attempts to get Google Container Registry authentication
+func getGCRAuth(registry string) (authn.Authenticator, error) {
+	// Check for Google Cloud credentials
+	if os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") != "" || os.Getenv("GCP_PROJECT") != "" {
```

Comment:
See other comment on bindenv

Reply:
Thanks for the feedback.
Addressed in next push.